pageserver/src/page_service.rs

```diff
@@ -790,6 +795,8 @@ impl PageServerHandler {
         }
     }
 
+    /// Handles the lsn lease request.
+    /// If a lease cannot be obtained, the client will receive NULL.
```

Comment:
Does 'lease cannot be obtained' always mean that the requested LSN has been GCed? Or there are other reasons why it can error out, e.g. some internal locks, etc.

I just worry now that we can stop retrying even though there is still a chance

Reply:
With the current implementation, it is the case that we explicitly return `NULL` if the requested LSN has been GC-ed. I think this is the only permanent error case.

As mentioned in point 1 of https://github.com/neondatabase/neon/issues/8817, to address some edge cases (pageserver live migration & restarts), we will stop doing GC cutoff validation for the page_service lease request. In that case, we might just keep retrying for the lease (_side note: this would require the initial request with validation to be made from control plane side using the HTTP API_).

I think I will keep this for now and refactor once I come up with a good plan to address #8817.

Reply:
Fair, thanks for the context